### PR TITLE
Remove workaround for old *feature bug and test hasFeature before using it (bug 1180909)

### PR DESCRIPTION
**Pull request by diox**
https://bugzilla.mozilla.org/show_bug.cgi?id=1180909

### CODE REVIEW

File: package/iframe/main.js [hunk 1]
```diff
@@ -54,56 +54,38 @@
...
+                    // that it's not empty, so we can just use the getFeature()
+                    // promise directly.
+                    promises.push(navigator.getFeature(key));
+                } else if (typeof window.hasFeature !== 'undefined') {
+                    // Regular boolean feature: just call hasFeature().
                     promises.push(navigator.hasFeature(key));
+                } else {
+                    // We should be using hasFeature() but it's absent, return
+                    // false.
+                    promises.push(false);
```
Comment (mstriemer) [inline package/iframe/main.js]: No need to wrap this in a promise?

File: package/iframe/main.js [hunk 2]
```diff
@@ -54,56 +54,38 @@
...
-                    }));
-                } else {
-                    // Regular case: just call hasFeature().
+            } else if (typeof key === 'string') {
+                if (key === 'acl.version') {
+                    // This feature has a value, so we need to call
+                    // getFeature(), but we actually only care about the fact
+                    // that it's not empty, so we can just use the getFeature()
+                    // promise directly.
+                    promises.push(navigator.getFeature(key));
```
Comment (mstriemer) [inline package/iframe/main.js]: Why don't we need to check for the existence of `navigator.getFeature` here?

Comment (diox) [inline package/iframe/main.js]: We do earlier, before doing all this mess

File: package/iframe/main.js [hunk 3]
```diff
@@ -54,56 +54,38 @@
...
+                } else if (typeof window.hasFeature !== 'undefined') {
```
Comment (mstriemer) [inline package/iframe/main.js]: This is checking for `hasFeature` on `window` but we use it from `navigator`.

File: package/iframe/main.js (hunk 1 above)
Comment (diox) [inline package/iframe/main.js]: No, `Promise.all()` does that for us. From MDN: 
> If something passed in the iterable array is not a promise, it's converted to one by Promise.resolve.

File: package/iframe/main.js (hunk 3 above)
Comment (diox) [inline package/iframe/main.js]: Ah, good catch.

File: package/iframe/main.js [hunk 4]
```diff
@@ -54,56 +54,38 @@
...
+                    // Regular boolean feature: just call hasFeature().
                     promises.push(navigator.hasFeature(key));
```
Comment (mstriemer) [inline package/iframe/main.js]: This seems to no longer handle the case for bug 1098470 (check `getFeature` after `hasFeature`). Is that intentional?

Comment (diox) [inline package/iframe/main.js]: Yes, that bug has been fixed since.

**Comment by mstriemer** [comment]
lgtm r+

**Pull request merged**